objects/resource.rb

```diff
@@ -47,6 +47,8 @@ def take(host = nil, port = nil)
         http.request(req)
       end
       [res.code.to_i, res.body, to_text(req, res)]
+    rescue Timeout::Error
+      [500, '', 'The request timed out after 5 seconds.']
```

Comment:
@pdacostaporto you're using number `5` here and the same number a few lines above. That's a code duplication. If in the future I change the number above, this one won't be changed and it will be a bug. It's better either to introduce a constant or get the real timeout value from the error